### PR TITLE
Refactor ExampleBrowser

### PR DESCRIPTION
**Pull request by kachkaev**
- navigate between examples with react-router
  (each example has got a slug)
- extract ExampleViewer into a separate component,
  determine its size using react-sizeme
- make ExampleBrowser component functional (i.e. pure)

react-three-renderer-routes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
     "cannon": "^0.6.2",
     "react": "~15.3.1",
     "react-dom": "^15.3.1",
+    "react-router": "^3.0.0-alpha.3",
```
Comment (kachkaev) [inline package.json:23]: using 3.0 instead of 2.\* because of https://github.com/reactjs/react-router/issues/3611#issuecomment-240950257

**Comment by toxicFork** [comment]
Thank you for the PR, I think I will probably go with https://github.com/toxicFork/react-three-renderer/pull/82 but have not had the time to make a decision yet

**Comment by kachkaev** [comment]
I agree that using the storybook would be much better. This PR can be a temporal patch until that idea has turned into code. I found it quite inconvenient not to have permalinks for examples while refreshing the app, so mainly fixing just this.

**Comment by toxicFork** [comment]
good point!

**Comment by kachkaev** [comment]
ping :–)

**Comment by toxicFork** [comment]
Hi, apologies for the lateness, I have been having issues with my internet
provider lately as I have just moved.

I will go to a coffee shop sometime this week with my laptop to merge, test
and push :D

Firtina

**Pull request merged**